DashAI/back/api/api_v1/endpoints/datasets.py

```diff
@@ -77,6 +85,70 @@ async def get_dataset(dataset_id: int, db: Session = Depends(get_db)):
     return dataset
 
 
+@router.get("/sample/{dataset_id}")
+async def get_sample(dataset_id: int, db: Session = Depends(get_db)):
+    """Return the dataset with id dataset_id from the database.
+
+    Parameters
+    ----------
+    dataset_id : int
+        id of the dataset to query.
+
+    Returns
+    -------
+    Dict
+        A Dict with a sample of 10 rows
+    """
+    try:
+        file_path = db.get(Dataset, dataset_id).file_path
+        dataset: DashAIDataset = load_dataset(f"{file_path}/dataset")
+        sample = dataset["train"].sample(n=10)
+        if not file_path:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail="Dataset not found",
+            )
+    except exc.SQLAlchemyError as e:
+        log.exception(e)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal database error",
+        ) from e
+
+    return sample
+
+
+@router.get("/types/{dataset_id}")
+async def get_types(dataset_id: int, db: Session = Depends(get_db)):
+    """Return the dataset with id dataset_id from the database.
+
+    Parameters
+    ----------
+    dataset_id : int
+        id of the dataset to query.
+
+    Returns
+    -------
+    Dict
+        Dict containing column names and types.
+    """
+    try:
+        file_path = db.get(Dataset, dataset_id).file_path
+        column_types = get_column_types(f"{file_path}/dataset")
+        if not column_types:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail="Dataset not found",
```

Comment:
In this case, the error message should indicate the current error:

```suggestion
                detail="Column types not found", # or error while loading column types.
```